mobile/src/navigation/app/common/helpers/group-by-key/group-by-key.helper.ts

```diff
@@ -0,0 +1,18 @@
+import { DrawerNavigationItem } from '../../types/types';
+
+const groupByKey = (
+  items: DrawerNavigationItem[],
+  key: keyof DrawerNavigationItem,
+): DrawerNavigationItem[][] => {
```

Comment:
It should be a common helper, move it to src/helpers/object/group-by-key also pass item type as generic. Also return type should be an object
```suggestion
const groupByKey = <T>(
  items: T[],
  key: keyof T,
): Record<T[keyof T], items[]> => {
```